File: creditcarddesign/src/main/java/com/cooltechworks/creditcarddesign/pager/CreditCardFragment.java
package com.cooltechworks.creditcarddesign.pager;

import android.support.v4.app.Fragment;
import android.text.TextWatcher;

/**
 * Created by sharish on 9/1/15.
 */
public abstract  class CreditCardFragment extends Fragment implements TextWatcher, IFocus {

    protected IActionListener mActionListener;

    public void setActionListener(IActionListener listener) {
        mActionListener = listener;
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {

    }


    public void onEdit(String edit) {

        if(mActionListener != null) {
            mActionListener.onEdit(this,edit);
        }

    }

    public void onComplete() {

        if(mActionListener != null) {
            mActionListener.onActionComplete(this);
        }

    }

}

File: creditcarddesign/src/main/java/com/cooltechworks/creditcarddesign/pager/CreditCardFragment.kt
package com.cooltechworks.creditcarddesign.pager

import androidx.fragment.app.Fragment
import android.text.TextWatcher

/**
 * Created by sharish on 9/1/15.
 */
abstract class CreditCardFragment : Fragment(), TextWatcher, IFocus {

    protected var mActionListener: IActionListener? = null

    fun setActionListener(listener: IActionListener) {
        mActionListener = listener
    }

    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

    }

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

    }


    fun onEdit(edit: String) {

        if (mActionListener != null) {
            mActionListener!!.onEdit(this, edit)
        }

    }

    fun onComplete() {

        if (mActionListener != null) {
            mActionListener!!.onActionComplete(this)
        }

    }

}
